lldb-eval/eval.cc

```diff
@@ -269,8 +311,42 @@ void Interpreter::Visit(const LiteralNode* node) {
 }
 
 void Interpreter::Visit(const IdentifierNode* node) {
-  Value val =
-      static_cast<const Context::IdentifierInfo&>(node->info()).GetValue();
+  auto identifier = static_cast<const Context::IdentifierInfo&>(node->info());
+
+  Value val;
+  switch (identifier.kind()) {
+    using Kind = Context::IdentifierInfo::Kind;
+    case Kind::kValue:
+      val = identifier.value();
+      assert(val.IsValid() && "invalid ast: invalid identifier value");
+      break;
+    case Kind::kContextVar:
+      assert(node->is_context_var() && "invalid ast: context var expected");
+      val = ResolveContextVar(node->name());
+      if (!val.IsValid()) {
```

Comment:
This can only happen if the expresion was evaluated with a different set of context variables, right?

Reply:
Yes.

Once we split parsing and evaluation, it may also happen that parsing and evaluation parts are provided with the same-name context variables, but different types. Added a TODO to check that.